packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+    parentEntity: string;
+    childType: string;
+    variables: string[];
+    date?: string;
+    facetIds?: string[];
+  }): Promise<PointApiResponse> {
+    const queryString = toURLSearchParams({
+      parentEntity: params.parentEntity,
+      childType: params.childType,
+      variables: params.variables,
+      date: params.date,
+      facetIds: params.facetIds,
+    });
+    const url = `${
+      this.apiRoot || ""
+    }/api/observations/point/within?${queryString}`;
+    return (await (await fetch(url)).json()) as PointApiResponse;
+  }
+
+  /**
+   * Gets the data from /api/observations/series endpoint.
+   * @param entities list of enitites to get data for
+   * @param variables list of variables to get data for
+   */
+  async getObservationsSeries(params: {
+    entities: string[];
+    variables: string[];
+    facetIds?: string[];
+  }): Promise<SeriesApiResponse> {
+    const queryString = toURLSearchParams({
+      entities: params.entities,
+      variables: params.variables,
+      facetIds: params.facetIds,
+    });
+    const url = `${this.apiRoot || ""}/api/observations/series?${queryString}`;
+    return (await (await fetch(url)).json()) as SeriesApiResponse;
+  }
+
+  /**
+   * Gets the data from /api/observations/series/within endpoint.
+   * @param parentEntity parent place to get the data for
+   * @param childType place type to get the data for
+   * @param variables variables to get data for
+   * @returns
+   */
+  async getObservationsSeriesWithin(params: {
+    parentEntity: string;
+    childType: string;
+    variables: string[];
+    facetIds?: string[];
+  }): Promise<SeriesApiResponse> {
+    const queryString = toURLSearchParams({
+      parentEntity: params.parentEntity,
+      childType: params.childType,
+      variables: params.variables,
+      facetIds: params.facetIds,
+    });
+    const url = `${
+      this.apiRoot || ""
+    }/api/observations/series/within?${queryString}`;
+    return (await (await fetch(url)).json()) as SeriesApiResponse;
+  }
+
+  /**
+   * Fetches node properties from the provided list of dcids
+   * @param dcids node dcids
+   * @param props properties to fetch
+   * @returns
+   */
+  private async getNodePropValues(
+    dcids: string[],
+    props: string[]
+  ): Promise<NodePropValues> {
+    if (dcids.length === 0 || props.length === 0) {
+      return {};
+    }
+    const nodePropValues: NodePropValues = {};
+    for (const propName of props) {
+      nodePropValues[propName] = await this.getFirstNodeValues({
+        dcids,
+        prop: propName,
+      });
+    }
+    return nodePropValues;
+  }
+
+  /**
+   * Find the observation with the closest date to targetDate
+   * @param observations sorted observations
+   * @param targetDate date string
+   */
+  private getClosestObservationToDate(
+    observations: Observation[],
+    targetDate: string
+  ): Observation | undefined {
+    const index = _.sortedIndexBy(
```

Comment:
Should either handle empty list for observations and empty string for targetDate or have a comment that the two parameters can't be empty

Reply:
Good catch- added a handler for when targetDate is blank. It looks like the empty observation case already works, as indexing out of bounds in javascript will return `undefined` which is the return value i want (`[][0] === undefined`)